### PR TITLE
Group routes without a prefix.

### PR DESCRIPTION
**Pull request by francis94c**
Each pull request should address a single issue and have a meaningful title.

**Description**
I ran into bit of an issue while grouping my routes and wanted to achieve something like the below.
```
$routes->group('v1', ['namespace' => 'App\Controllers\V1'], function ($routes) {
    $routes->post('verify/begin', 'VerificationController::generateVerificationCode');
    $routes->post('verify/code', 'VerificationController::verifyCode');
    $routes->post('users', 'UserController::createUser');
    $routes->post('auth', 'AuthController::authenticateUser');
    
    $routes->group(null, ['filter' => 'api-auth', 'namespace' => 'App\Controllers\V1'], function($routes) {
        $routes->put('users', 'UserController::updateUser');
    });
});
```
In essence i wanted to group some routes without a prefix but have them run under certain filters. I realized that when i hit the below end point
```
PUT /v1/users
```
I basically get a 404.

So  i made some changes to make this possible.

These changes could help one group routes under a subdomain for instance without the need for a route prefix, apply security filters on certain routes like i did above, without the need for a prefix, and so on.

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPdocs
- [x] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Comment by paulbalandan** [comment]
Is the same not achieved by using an empty string instead of `null`?

```php
    $routes->group('', ['filter' => 'api-auth', 'namespace' => 'App\Controllers\V1'], function($routes) {
        $routes->put('users', 'UserController::updateUser');
    });
```

**Comment by francis94c** [comment]
That's what i thought at first, I tried it but it didn't work.

On investigating the `RouteCollection` class.

on line 739, i see 

```
$this->group = ltrim($oldGroup . '/' . $name, '/');
```

With an empty string, we are left with '{oldGroup}//' which won't match.

I replaced the line with
```
$this->group = $name ? ltrim($oldGroup . '/' . $name, '/') : $oldGroup;
```
to make it work...

**Comment by paulbalandan** [comment]
I think it is a bug. Your added line would also work for empty strings as it will evaluate to `false`.

> I replaced the line with
> 
> ```
> $this->group = $name ? ltrim($oldGroup . '/' . $name, '/') : $oldGroup;
> ```

So maybe instead of going for a nullable string typehint, we stick to string typehint and just add that line?

I'll have the others say what they think on this. @MGatner ?

**Comment by francis94c** [comment]
I'll update and push

File: system/Router/RouteCollection.php [hunk 1]
```diff
@@ -713,7 +713,8 @@ public function group(string $name, ...$params)
 
 		// To register a route, we'll set a flag so that our router
 		// so it will see the group name.
-		$this->group = ltrim($oldGroup . '/' . $name, '/');
+		// If the group name is null, we go on using the previously built group name.
```
Comment (paulbalandan) [inline system/Router/RouteCollection.php:716]:
```suggestion
		// If the group name is empty, we go on using the previously built group name.
```

**Review by paulbalandan — APPROVED** [review]

**Comment by paulbalandan** [comment]
@francis94c thank you for your first contribution

**Comment by francis94c** [comment]
My pleasure @paulbalandan

**Pull request merged**